runtime/switchboard_impl.hpp

```diff
@@ -334,7 +334,7 @@ namespace ILLIXR {
 			  - Calls topic.schedule, which acquires _m_callbacks_lock, (see its proof of thread-safety)
 			  Therefore this method is thread-safe.
 			 */
-			const std::lock_guard lock{_m_registry_lock};
+			const std::shared_lock lock{_m_registry_lock};
```

Comment:
`try_emplace` is an operation that could trigger a write (if the key does not already exist. Therefore, we have to use `unique_lock` before calling `_m_registry.try_emplace`.